packages/cozy-sharing/src/queries/queries.js

```diff
@@ -59,10 +59,52 @@ export const buildReachableContactsQuery = () => ({
   }
 })
 
-export const buildContactGroupsQuery = () => ({
-  definition: Q(Group.doctype),
+export const buildContactGroupsByIdsQuery = ids => ({
+  definition: Q(Group.doctype).getByIds(ids),
   options: {
-    as: 'io.cozy.contacts.groups'
+    as: `io.cozy.contacts.groups/by-ids/${ids.join('/')}`,
+    fetchPolicy: defaultFetchPolicy,
+    enabled: ids.length > 0
+  }
+})
+
+export const buildUnreachableContactsFromGroupsQuery = groupIds => ({
```

Comment:
@paultranvan I'd like your opinion on this query. I've used `$elemMatch` which works well locally but I'm wondering about performance. For a bit more context, we want to retrieve all the contact groups with contactable members as well as their non-contactable members.

I've opted for a two-stage approach: 
1. I retrieve all the contacts that can be contacted. This request allows me to retrieve their group ids and is used in another part of the interface.
2. I retrieve the groups via their ids as well as their non-contactable contacts via this query. 

I had considered another option, which was to retrieve all the contacts and all the groups in order to do the same processing on the front end, but the request can be quite large, so I tried to keep the initial optimisations.

Reply:
We discuss together and we decide to search all unreachable that have a group to avoir having dynamic variable in a partial index